### PR TITLE
MULE-11701: Remove custom-source element

### PR DESCRIPTION
**Pull request by lucianoRM**

### CODE REVIEW

**Comment by lucianoRM** [comment]
I've created a different issue for that and i'm working on it in a different branch: MULE-12975

File: tests/component-plugin/src/main/java/org/mule/functional/api/component/SkeletonSource.java [hunk 1]
```diff
@@ -15,9 +17,10 @@
  *
  * @since 4.0
  */
-public class SkeletonSource extends AbstractAnnotatedObject implements MessageSource {
+public class SkeletonSource extends AbstractAnnotatedObject implements MessageSource, Startable {
 
   private Processor listener;
+  private boolean started;
```
Comment (elrodro83) [inline tests/component-plugin/src/main/java/org/mule/functional/api/component/SkeletonSource.java]: make volatile

File: tests/component-plugin/src/main/java/org/mule/functional/config/TestComponentBuildingDefinitionProvider.java [hunk 2]
```diff
@@ -149,6 +149,9 @@ public void init() {
         .withTypeDefinition(fromType(SharedConfig.class))
         .build());
 
+    componentBuildingDefinitions.add(baseDefinition.copy().withIdentifier("custom-source")
```
Comment (elrodro83) [inline tests/component-plugin/src/main/java/org/mule/functional/config/TestComponentBuildingDefinitionProvider.java]: remove

File: tests/component-plugin/src/main/resources/META-INF/mule-test.xsd [hunk 3]
```diff
@@ -301,4 +301,25 @@
         </xsd:complexContent>
     </xsd:complexType>
 
+    <xsd:element name="custom-source" type="testMessageSourceType" substitutionGroup="mule:abstract-message-source">
```
Comment (elrodro83) [inline tests/component-plugin/src/main/resources/META-INF/mule-test.xsd]: remove

**Pull request merged**